### DISCUSSION TITLE
Making silx and Optional Dependency

### DISCUSSION THREAD

**Discussion by CSSFrancis** [silx-kit/pyFAI#2058] [Q&A] [answered]
What would be involved in making `Silx` an optional dependency for pyfai?

We make use of pyfai downstream in pyxem (https://github.com/pyxem/pyxem) for doing azimuthal integration on electron diffraction patterns but many of the extra `Silx` related interactive features aren't necessary. 

I haven't looked too closely into the pyfai code but my understanding is that most of the caking algorithms should be able to stand on their own without `Silx`.

Thanks!

**Comment by kif** [comment, marked as answer]
Hi Francis,

Thanks to let me know pyFAI is used in the electron diffraction community. Some physical constants are radically different for photon and electron. If you find some part of the code where there is a too strong assumption that we are dealing with photons, please let me know.

I understand you concern about dependencies, but I am sorry: `pyFAI` without `silx` is not an option for now ... let me give you a few examples:
- The main GUI `pyFAI-calib2` is relying on Qt widgets developed by the silx team and co-developped with them.
- The testing of pyFAI is using the ExternalResources from silx to provide test files (downloaded and cached locally on the disk ...)
- The OpenCL part of pyFAI, i.e. all GPU computing relies on some initialization provided by `silx.opencl`
Since the requirements for silx is not (much) more than the one for pyFAI, I don't see a valid reason to duplicate several thousands of line of code (meaning much more maintenance as well).

Cheers,

Jerome
